src/main/java/com/example/sortmanager/SortServiceManager.java

```diff
@@ -0,0 +1,72 @@
+package com.example.sortmanager;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.os.Build;
+import android.os.IBinder;
+import android.os.RemoteException;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
+
+import com.example.sortingapp.ISortService;
+import com.example.sortingapp.SortingMethod;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+
+public class SortServiceManager {
+    private static final long AWAIT_TIMEOUT_MILLS = 1000;
+
+    private CountDownLatch mLatch = new CountDownLatch(1);
+    private ISortService mSortService = null;
```

Comment:
```suggestion
    private volatile ISortService mSortService = null;
```

Create Trello ticket with "Learn about volatile fields"